animation_nodes/nodes/spline/splines_from_edges_utils.pyx

```diff
@@ -0,0 +1,113 @@
+from ... data_structures cimport (
+    Mesh, Vector3DList, EdgeIndicesList, PolySpline,
+    VirtualDoubleList, FloatList, IntegerList
+)
+
+def splinesFromBranches(Vector3DList vertices, EdgeIndicesList edges, VirtualDoubleList radii):
+    cdef int i, j
+    cdef int edgesAmount = edges.length
+    cdef int verticesAmount = vertices.length
+
+    # Compute how many neighbour each vertex have.
+    cdef IntegerList neighboursAmounts = IntegerList.fromValue(0, length = verticesAmount)
+    for i in range(edgesAmount):
+        neighboursAmounts.data[edges.data[i].v1] += 1
+        neighboursAmounts.data[edges.data[i].v2] += 1
+
+    # Compute the start index of each group of neighbours of each vertex.
+    cdef IntegerList neighboursStarts = IntegerList(length = verticesAmount)
+    cdef int start = 0
+    for i in range(verticesAmount):
+        neighboursStarts.data[i] = start
+        start += neighboursAmounts.data[i]
+
+    # Keep track of how many index is in each group of neighbours at each iteration.
+    cdef IntegerList filledSpaces = IntegerList.fromValue(0, length = verticesAmount)
+
+    # Compute the indices of neighbouring vertices of each vertex.
+    cdef IntegerList neighbours = IntegerList(length = edgesAmount * 2)
+    for i in range(edgesAmount):
+        neighbours.data[neighboursStarts.data[edges.data[i].v1] +
```

Comment:
I'd suggest extra `v1` and `v2` variables of type `unsigned int` to reduce line length and improve readability.

Reply:
I think in this case it is fine. You could just check if cython generates efficient code for it by checking the generated c code.
(You can create an annotated html version of the file with this command: `cython -a path/to/file.pyx`)